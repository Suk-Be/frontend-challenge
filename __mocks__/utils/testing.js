import { unmountComponentAtNode } from 'react-dom';

export const isolateRenderTest = (containerElement) => {
	beforeEach(() => {
		// setup a DOM element as a render target
		container = document.createElement('div');
		document.body.appendChild(containerElement);
	});

	afterEach(() => {
		// cleanup on exiting
		unmountComponentAtNode(containerElement);
		containerElement.remove();
		containerElement = null;
	});
};

export const isolateTimerTest = (containerElement) => {
	beforeEach(() => {
		// setup a DOM element as a render target
		containerElement = document.createElement("div");
		document.body.appendChild(containerElement);
		jest.useFakeTimers();
	  });
	  
	  afterEach(() => {
		// cleanup on exiting
		unmountComponentAtNode(containerElement);
		containerElement.remove();
		containerElement = null;
		jest.useRealTimers();
	  });

}