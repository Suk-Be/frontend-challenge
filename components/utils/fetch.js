export const githubURL = `https://api.github.com/search/repositories?q=created:%3E2017-01-10&sort=stars&order=desc`;

export const fetchData = async (url) => {
	const res = await fetch(url);
	const data = await res.json();

	if (!data) {
		return {
			notFound: true,
		};
	}

	return data;
};
