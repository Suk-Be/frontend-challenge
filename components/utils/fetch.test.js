import { fetchData, githubURL } from './fetch';
import { itemsMock } from '../../__mocks__/utils/dataMock';

const unmockedFetch = global.fetch;

beforeAll(() => {
	global.fetch = () =>
		Promise.resolve({
			json: () => Promise.resolve(itemsMock),
		});
});

afterAll(() => {
	global.fetch = unmockedFetch;
});

describe('fetchData', () => {
	test('fetches data', async () => {
		const json = await fetchData();
		expect(Array.isArray(json)).toEqual(true);
	});
    test("fetch the same url", () => {
        const url = githubURL
        expect(url).toMatchSnapshot();
    })
});
