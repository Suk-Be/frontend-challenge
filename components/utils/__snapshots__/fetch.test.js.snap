// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchData fetch the same url 1`] = `"https://api.github.com/search/repositories?q=created:%3E2017-01-10&sort=stars&order=desc"`;

exports[`fetchData fetche the same url 1`] = `"https://api.github.com/search/repositories?q=created:%3E2017-01-10&sort=stars&order=desc"`;
